src/utils/chart.ts

```diff
@@ -125,3 +127,73 @@ export const getFeeRateSamples = (feeRates: Array<FeeRateTracker.TransactionFeeR
 
   return samples
 }
+
+type AssertIsArray = (
+  value: EChartOption.Tooltip.Format | EChartOption.Tooltip.Format[],
+) => asserts value is EChartOption.Tooltip.Format[]
+
+export const assertIsArray: AssertIsArray = (value: EChartOption.Tooltip.Format | EChartOption.Tooltip.Format[]) => {
```

Comment:
Consider a simpler and more generic implementation:

```
assert(Array.isArray(dataList))
```
or
```
import { assert } from './error'

export function assertArray<T>(value: unknown | T[]): asserts value is T[] {
  assert(Array.isArray(value), 'value is not an array')
}
```
or
```
export function assertArray<T>(value: unknown | T[]): asserts value is T[] {
  if (!Array.isArray(value)) {
    throw new Error('value is not an array')
  }
}
```

Reply:
I think it's more generic, but not simple, cause we need to check if array item is real type of `T`, currently we only care about is Array or not, and it belongs to `utils/chart.ts` file, maybe refactor it to more generic way if we really need it in future.

Reply:
If you expect it to be a non-generic function, then it should not be named `assertIsArray`, but should have a more precise intent, such as `assertEChartFormatArray`.